k8s/migration/internal/controller/migration_controller.go

```diff
@@ -316,6 +341,105 @@ func newHostPathType(pathType string) *corev1.HostPathType {
+		return false, fmt.Errorf("OpenstackCreds '%s' CR is not validated", credsobj.(*vjailbreakv1alpha1.OpenstackCreds).Name)
+	}
+	return true, nil
+}
+
+//nolint:dupl // Similar logic to networks reconciliation, excluding from linting to keep it readable
+func reconcileStorage(ctx context.Context,
+	migration *vjailbreakv1alpha1.Migration,
+	vmwcreds *vjailbreakv1alpha1.VMwareCreds,
+	openstackcreds *vjailbreakv1alpha1.OpenstackCreds,
+	vm string,
+	storages []vjailbreakv1alpha1.Storage) ([]string, error) {
+	vmds, err := GetVMwDatastore(ctx, vmwcreds, migration.Spec.Source.DataCenter, vm)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get datastores: %w", err)
+	}
+
+	openstackvolumetypes := []string{}
+	for _, vmdatastore := range vmds {
+		for _, storagemaptype := range storages {
+			if vmdatastore == storagemaptype.Source {
+				openstackvolumetypes = append(openstackvolumetypes, storagemaptype.Target)
+			}
+		}
+	}
+	if len(openstackvolumetypes) != len(vmds) {
+		return nil, fmt.Errorf("VMware Datastore(s) not found in StorageMapping")
+	}
+
+	err = VerifyStorage(ctx, openstackcreds, openstackvolumetypes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to verify datastores: %w", err)
+	}
+	return openstackvolumetypes, nil
+}
+
+func reconcileMapping(ctx context.Context,
+	migration *vjailbreakv1alpha1.Migration,
+	openstackcreds *vjailbreakv1alpha1.OpenstackCreds,
+	vmwcreds *vjailbreakv1alpha1.VMwareCreds,
+	vm string,
+	networks []vjailbreakv1alpha1.Network,
+	storages []vjailbreakv1alpha1.Storage) (openstacknws, openstackvolumetypes []string, err error) {
+	openstacknws, err = reconcileNetwork(ctx, migration, openstackcreds, vmwcreds, vm, networks)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to reconcile network: %w", err)
+	}
+
+	openstackvolumetypes, err = reconcileStorage(ctx, migration, vmwcreds, openstackcreds, vm, storages)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to reconcile storage: %w", err)
+	}
+	return openstacknws, openstackvolumetypes, nil
+}
+
+//nolint:dupl // Similar logic to storages reconciliation, excluding from linting to keep it readable
+func reconcileNetwork(ctx context.Context,
+	migration *vjailbreakv1alpha1.Migration,
+	openstackcreds *vjailbreakv1alpha1.OpenstackCreds,
+	vmwcreds *vjailbreakv1alpha1.VMwareCreds,
+	vm string,
+	networks []vjailbreakv1alpha1.Network) ([]string, error) {
+	vmnws, err := GetVMwNetworks(ctx, vmwcreds, migration.Spec.Source.DataCenter, vm)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get network: %w", err)
+	}
+
+	openstacknws := []string{}
+	for _, vmnw := range vmnws {
+		for _, nwm := range networks {
+			if vmnw == nwm.Source {
+				openstacknws = append(openstacknws, nwm.Target)
+			}
+		}
+	}
+	if len(openstacknws) != len(vmnws) {
+		return nil, fmt.Errorf("VMware Network(s) not found in NetworkMapping")
+	}
+
+	err = VerifyNetworks(ctx, openstackcreds, openstacknws)
```

Comment:
It might be useful to update the `NetworkMapping` status here, before we try to use it in a `Migration`. Thoughts?

Reply:
Fixed

Reply:
status:
      networkMappinValidationMessage: NetworkMapping validated
      networkMappingValidationStatus: Succeeded